src/App.tsx

```diff
@@ -15,36 +17,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const sortingActions = {
+  alphabetically: 'Alphabetically',
+  length: 'Length',
+};
+
+function handleGoodsSortAction(
+  goods: string[],
+  action: string,
+  isReversed: boolean,
+) {
+  const renderedGoods = [...goods];
+
+  renderedGoods.sort((good1, good2) => {
+    switch (action) {
+      case sortingActions.alphabetically:
+        return good1.trim().localeCompare(good2.trim());
+      case sortingActions.length:
+        return good1.trim().length - good2.trim().length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    renderedGoods.reverse();
+  }
+
+  return renderedGoods;
+}
+
+function resetState<T extends unknown[]>(
+  ...args: { [K in keyof T]: { setter: StateSetter<T[K]>; initialValue: T[K] } }
+): void {
+  args.forEach(({ setter, initialValue }) => setter(initialValue));
+}
+
 export const App: React.FC = () => {
+  const [sortingAction, setSortingAction] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const { alphabetically, length } = sortingActions;
+
+  const handledGoods = handleGoodsSortAction(
+    goodsFromServer,
+    sortingAction,
+    reversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
```

Comment:
[UX] - It's a good practice to mark the active button to explain to the user what action is active. However, the current implementation might not be clear enough as it only changes the button style. Consider adding an explicit visual cue, like an icon or text, to indicate the currently active sorting action.

Reply:
not my styes